### PR TITLE
Fix :attr:`.ManimConfig.format` not updating movie file extension

### LINKED ISSUE #3837: Suffix of generated video file is incorrect when passing `--format`
## Reproduce
```py
from manim import *
class Manimation(Scene):
    def construct(self):
        self.play(Create(Circle()))
with tempconfig({"format", "webm"}):
    Manimation().render()
```
This will produce a file called `Manim.mp4`. Upon inspection with ffprobe I get
<details><summary>
ffprobe output
</summary>

```
ffprobe version n7.0.1 Copyright (c) 2007-2024 the FFmpeg developers
  built with gcc 14.1.1 (GCC) 20240522
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-amf --enable-avisynth --enable-cuda-llvm --enable-lto --enable-fontconfig --enable-frei0r --enable-gmp --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdav1d --enable-libdrm --enable-libdvdnav --enable-libdvdread --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libharfbuzz --enable-libiec61883 --enable-libjack --enable-libjxl --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libplacebo --enable-libpulse --enable-librav1e --enable-librsvg --enable-librubberband --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpl --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-libzimg --enable-mbedtls --enable-nvdec --enable-nvenc --enable-opencl --enable-opengl --enable-shared --enable-vapoursynth --enable-version3 --enable-vulkan
  libavutil      59.  8.100 / 59.  8.100
  libavcodec     61.  3.100 / 61.  3.100
  libavformat    61.  1.100 / 61.  1.100
  libavdevice    61.  1.100 / 61.  1.100
  libavfilter    10.  1.100 / 10.  1.100
  libswscale      8.  1.100 /  8.  1.100
  libswresample   5.  1.100 /  5.  1.100
  libpostproc    58.  1.100 / 58.  1.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/home/jasongrace2282/repos/manim/media/videos/1080p60/manim.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf60.16.100
    comment         : Rendered with Manim Community v0.18.1
  Duration: 00:00:01.00, start: 0.000000, bitrate: 105 kb/s
  Stream #0:0[0x1](und): Video: vp9 (Profile 0) (vp09 / 0x39307076), yuv420p(tv), 1920x1080, 96 kb/s, 60 fps, 60 tbr, 90k tbn (default)
      Metadata:
        handler_name    : VideoHandler
        vendor_id       : [0][0][0][0]
```

</details>
Which - as far as I can tell - looks to be a `.mp4` file.

## The Cause
`ManimConfig.resolve_movie_file_extension` appears to only be called when the `config.transparent` option is set. This should be changed to accommodate for just `format` changing.

### PR DESCRIPTION
**Pull request by JasonGrace2282**
Closes #3837

## Example of Bug
```pycon
>>> from manim import config
>>> config.movie_file_extension
'.mp4'
>>> config.format = "webm"
>>> config.movie_file_extension
'.mp4'
>>> config.transparent = config.transparent
>>> config.movie_file_extension
'.webm'
```

### CODE REVIEW

**Review by behackl — APPROVED** [review]
LGTM

**Comment by behackl** [comment]
should we add a test for this?

**Comment by JasonGrace2282** [comment]
Good idea, I'm not sure how I forgot that :)

File: tests/test_config.py [hunk 1]
```diff
@@ -33,6 +34,20 @@ def test_tempconfig(config):
             assert config[k] == v
 
 
+@pytest.mark.parametrize(
+    ("format", "expected_file_extension"),
+    [
+        ("mp4", ".mp4"),
+        ("webm", ".webm"),
+        ("mov", ".mov"),
+        ("gif", ".mp4"),
```
Comment (behackl) [inline tests/test_config.py:43]: It is somewhat odd how differently we are treating the case of `.gif`, but that's a different discussion.

Comment (JasonGrace2282) [reply tests/test_config.py:43]: I think it's just a missed edge case, probably worth fixing (wow we have a lot of problems with `ManimConfig` xD)
